control-base/src/main/java/fi/nls/oskari/control/layer/DataProviderHandler.java

```diff
@@ -105,4 +115,78 @@ private void deleteDataProvider(List<OskariLayer> layers, boolean deleteLayers,
 	private void flushLayerListCache() {
 		CacheManager.getCache(GetMapLayerGroupsHandler.CACHE_NAME).flush(true);
 	}
+
+	public void handlePut(ActionParameters params) throws ActionException {
+        handlePost(params);
+    }
+    @Override
+    public void handlePost(ActionParameters params) throws ActionException {
+
+        final HttpServletRequest request = params.getRequest();
+        final DataProvider dataProvider = new DataProvider();
+        // hierarchical admin frontend sends separate params
+        dataProvider.setId(params.getHttpParam(PARAM_ID, -1));
+        dataProvider.setNames(RequestHelper.parsePrefixedParamsMap(request, PARAM_NAME_PREFIX));
+
+
+        if (dataProvider.getLocale() == null) {
+            // the classical layer admin sends JSON payload
+            // locale as payload  { id: "124", locales: {en: "name en", fi: "name fi"}}
+			try {
+				JSONObject payload = params.getPayLoadJSON();
+				dataProvider.setId(ConversionHelper.getInt(payload.optString("id"), -1));
+				JSONObject locales;
+				locales = payload.getJSONObject("locales");
+
+				JSONObject defaultLang = locales.optJSONObject(PropertyUtil.getDefaultLanguage());
+				if (defaultLang == null) {
+					throw new ActionParamsException("No locale for default lang: " + PropertyUtil.getDefaultLanguage());
+				}
+				String name = defaultLang.optString("name");
+				if (name == null || name.trim().isEmpty()) {
+					throw new ActionParamsException("No name for default lang: " + PropertyUtil.getDefaultLanguage());
+				}
+
+				dataProvider.setLocale(locales);
+			} catch (JSONException e) {
+				throw new ActionException("Cannot poulate maplayer group from request", e);
+			}
+        }
```

Comment:
You can change this and assume it's the same kind of request that is sent to MapLayerGroupsHandler.handlePut()/post